docs/architecture/modernize-desktop/windows-migration.md

```diff
@@ -0,0 +1,527 @@
+
+The Windows Community Toolkit offers two types of controls: Wrapped Controls and
+Hosting Controls
+
+##### Wrapped Controls
+
+These wrapped controls wrap some UWP controls into Windows Forms or WPF
+controls, hiding UWP concepts for those developers. These controls are:
+
+* WebView and WebViewCompatible
+* InkCanvas and InkToolbar
+* MediaPlayerElement
+* MapControl
+
+Add the `Microsoft.Toolkit.Wpf.UI.Controls` package to your project, include the
+reference to the namespace and start using them.
+
+```xml
+<Window
+        ...
+        xmlns:uwpControls="clr-namespace:Microsoft.Toolkit.Wpf.UI.Controls;assembly=Microsoft.Toolkit.Wpf.UI.Controls">
+<Grid>
+    <Grid.RowDefinitions>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition Height="\*"/>
+    </Grid.RowDefinitions>
+    <uwpControls:InkToolbar TargetInkCanvas="{x:Reference Name=inkCanvas}"/>
+    <uwpControls:InkCanvas Grid.Row="1" x:Name="inkCanvas" />
+</Grid>
+```
+
+##### Hosting Controls
+
+The power of XAML Islands extends to the majority of inbox controls, third party
+controls, and custom control develop for UWP that can be integrated into Windows
+Forms and WPF as an "Island" of fully functional UI. The `WindowsXamlHost` control
+for WPF and Windows Forms allows doing this.
+
+For example, to use the `WindowsXamlHost` control in WPF, you just need to add
+the `Microsoft.Toolkit.Wpf.UI.XamlHost` package provided by Windows Community
+Toolkit.
+
+Once you have placed your `WindowsXamlHost` into your UI code, you just need to
+specify which UWP type you want to load and you can choose to use a wrapped
+control like a Button or a more complex one composed by several different
+controls, that are a custom UWP control.
+
+Here is a sample code snippet that shows how to add a UWP Button:
+
+```xml
+<Window
+        ...
+        xmlns:xamlhost="clr-namespace:Microsoft.Toolkit.Wpf.UI.XamlHost;assembly=Microsoft.Toolkit.Wpf.UI.XamlHost">
+
+<xamlhost:WindowsXamlHost x:Name="myUwpButton"
+                          InitialTypeName="Windows.UI.Xaml.Controls.Button" />
+```
+
+There is a clear recommendation on how to approach this and is better to have
+one single and bigger XAML Island containing a custom composite control than to
+have several islands with one control on each.
+
+One of the core features of XAML is binding and it works between your Win32 code
+and the island so you can bind, for instance, a Win32 Textbox with a UWP
+Textbox. One important thing to consider is that these bindings are one-way
+bindings, from UWP to Win32, so if you update the Textbox inside the XAML Island
+the Win32 Textbox will be updated, but not the other way around.
+
+This link contains a walkthrough about how to use XAML Islands:
+
+<https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/host-standard-control-with-xaml-islands>
+
+#### Adding UWP XAML custom controls
+
+A XAML custom control is a control (or user control) created by you or by third
```

Comment:
third parties is weird. Usually third-party is an adjective like third-party companies, vendors, etc.